libimage/push.go

```diff
@@ -85,5 +87,21 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 
 	defer c.close()
 
-	return c.copy(ctx, srcRef, destRef)
+	manifestBytes, err := c.copy(ctx, srcRef, destRef)
+	if err != nil {
+		return nil, err
+	}
+	manifestDigest, err := manifest.Digest(manifestBytes)
+	if err != nil {
+		return nil, err
+	}
+	// Tag for pushed remote digest i.e repo@remote-digest
+	remoteDigest := strings.TrimPrefix(manifestDigest.String(), "sha256:")
+	pushedTag := destRef.DockerReference().Name() + ":" + remoteDigest
```

Comment:
Image references should always be manipulated using c/image/docker/reference; any string manipulation is suspect (and most of it, although I don’t think this one, gets corner cases wrong).

I’d expect something like `reference.WithDigest(reference.TrimNamed(ref), digest)`.